lib/js_rails_routes/generator.rb

```diff
@@ -57,15 +62,28 @@ def handle_route(route_name, route_path)
       "export function #{route_name}_path(params) { return process('#{route_path}', params, [#{keys.join(',')}]); }"
     end
 
-    def routes
-      @routes ||= Rails.application.routes.routes
-                       .select(&:name)
-                       .map { |r| [r.name, r.path.spec.to_s.split('(')[0]] }
-                       .sort { |a, b| a[0] <=> b[0] }
+    def routes_with_engines
+      engines = Rails::Engine.subclasses.reject { |klass| klass.to_s.downcase =~ exclude_engines }
+
+      engine_routes = engines.map do |engine|
+        routes = make_routes(engine.routes.routes)
+        [engine.to_s, routes]
+      end
+      engine_routes = engine_routes.reject { |_, routes| routes.empty? }
+      routes = engine_routes.push(['Rails', make_routes(Rails.application.routes.routes)])
+
+      @routes ||= routes
+    end
+
+    def make_routes(routes)
+      routes
+        .select(&:name)
+        .map { |r| [r.name, r.path.spec.to_s.split('(')[0]] }
+        .sort { |a, b| a[0] <=> b[0] }
     end
 
-    def write(string)
-      File.open(path, 'w') { |f| f.write(string) }
+    def write(klass, string)
+      File.open(File.join(base_path, "#{klass.gsub('::Engine', '').downcase}-routes.js"), 'w') { |f| f.write(string) }
```

Comment:
`klass` tends to be used for an instance of `Class` class, but in this case it is a `String` here. Please rename it as `rails_engine_name` for readability.

Reply:
Make a good point. :+1: